### PR TITLE
SSMapping

### PR DESCRIPTION
**Pull request by AffectedArc07**
**What does this PR do:**
This PR takes mapping elements out of the master controller (PS) and into its own subsystem, running off of StonedMC. This will allow it to be easier modified, and will allow for future plans such as map rotation, and lavaland.

**Changelog:**
:cl: AffectedArc07
add: ssMapping
tweak: Gateway and space ruins are now loaded with ssMapping
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   	name = "Mapping"
+   	init_order = 12
+   	flags = SS_NO_FIRE
+
+/datum/controller/subsystem/mapping/New()
+   	NEW_SS_GLOBAL(SSmapping)
+   	return ..()
+
+/datum/controller/subsystem/mapping/Initialize(timeofday)
+   	// Load all Z level templates
```
Comment (AffectedArc07) [inline code/controllers/subsystem/mapping.dm]: Dont know why this is indented weird, its not like that in my file
image

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
+SUBSYSTEM_DEF(mapping)
+   	name = "Mapping"
+   	init_order = 12
```
Comment (Citinited) [inline code/controllers/subsystem/mapping.dm]: I think you should be using a define for this (see the other subsystems)

Also, `INIT_ORDER_JOBS` is already defined as 12, not sure if that'll break anything but it's probably best to select a different value

Comment (AffectedArc07) [reply code/controllers/subsystem/mapping.dm]: Noted, will work on these when I am home from starbucks

**Comment by AffectedArc07** [comment]
Init order is now using the define for mapping, and the defines have actually been ordered properly.

**Comment by Kyep** [comment]
I'm going to wait for a maintainer to give their opinion, before I give mine on this.
Primarily as I'm not certain about what the technical implementations would be.

File: code/controllers/subsystem/mapping.dm (hunk 1 above)
Comment (Fox-McCloud) [reply code/controllers/subsystem/mapping.dm]: this file is literally all spaced indentation.

File: code/controllers/master_controller.dm [hunk 3]
```diff
@@ -36,29 +36,7 @@ var/global/pipe_processing_killed = 0
...
-
-	space_manager.do_transition_setup()
-
 	setupfactions()
 	setup_economy()
 
-	for(var/i=0, i<max_secret_rooms, i++)
-		make_mining_asteroid_secret()
-
 	populate_spawn_points()
```
Comment (Fox-McCloud) [inline code/controllers/master_controller.dm]: Oh, Master controller, you used to run everything....now uh, you're probably going to get killed off this year.

Comment (MrMagolor) [reply code/controllers/master_controller.dm]: I wonder how Goon feels that their fancy-shmancy process scheduler is getting killed off by tg's thingamabob

File: code/controllers/subsystem/mapping.dm [hunk 4]
```diff
@@ -0,0 +1,38 @@
+SUBSYSTEM_DEF(mapping)
+   	name = "Mapping"
+   	init_order = INIT_ORDER_MAPPING // 9
+   	flags = SS_NO_FIRE
+
+/datum/controller/subsystem/mapping/New()
+	NEW_SS_GLOBAL(SSmapping)
```
Comment (Fox-McCloud) [inline code/controllers/subsystem/mapping.dm]: Um. This is making the same SS twice.

Comment (AffectedArc07) [reply code/controllers/subsystem/mapping.dm]: This is exactly what TG has, but I will remove it if its bad

File: code/__DEFINES/subsystems.dm [hunk 5]
```diff
@@ -45,39 +45,39 @@
...
+#define INIT_ORDER_TICKER 10
+#define INIT_ORDER_TRAITS 11
+#define INIT_ORDER_JOBS 12
+#define INIT_ORDER_EVENTS 13
+#define INIT_ORDER_RESEARCH 14
+#define INIT_ORDER_INPUT 15
+#define INIT_ORDER_SERVER_MAINT 16
+#define INIT_ORDER_BLACKBOX 17
+#define INIT_ORDER_DBCORE 18
+#define INIT_ORDER_GARBAGE 19
```
Comment (Fox-McCloud) [inline code/__DEFINES/subsystems.dm]: Are you sure about that: https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/subsystems.dm#L53

File: code/controllers/master_controller.dm (hunk 3 above)
Comment (Fox-McCloud) [reply code/controllers/master_controller.dm]: To be honest, the biggest argument for Goon going open source is so they can use StonedMC, IMO.

Then they don't have to do anything silly like their "real time reagent/breathing/mob processing".

File: code/controllers/subsystem/mapping.dm (hunk 4 above)
Comment (Fox-McCloud) [reply code/controllers/subsystem/mapping.dm]: Very very old implementation of SS's, well before SS defines became a thing.

File: code/__DEFINES/subsystems.dm (hunk 5 above)
Comment (AffectedArc07) [reply code/__DEFINES/subsystems.dm]: Want me to change the orders of these to be TGs?

Comment (Fox-McCloud) [reply code/__DEFINES/subsystems.dm]: Might as well. As always, test to make sure nothing is completely broken.

Comment (AffectedArc07) [reply code/__DEFINES/subsystems.dm]: I will, though its hard to test **EVERY** SS, especially since we have some they dont, like mentor SS

Comment (Fox-McCloud) [reply code/__DEFINES/subsystems.dm]: Fair enough.

**Comment by Fox-McCloud** [comment]
What could possibly go wrong?

**Comment by KasparoVy** [comment]
OmG tEh WhOeL uNiVeRs iS gOn!1!one

**Pull request merged**